internal/planner/create.go

```diff
@@ -56,76 +56,65 @@ func (n *createNode) Kind() string { return "createNode" }
 
 func (n *createNode) Init() error { return nil }
 
-func (n *createNode) Start() error {
-	doc, err := client.NewDocFromMap(n.input, n.collection.Definition())
-	if err != nil {
-		n.err = err
-		return err
+func docIDsToSpans(ids []string, desc client.CollectionDescription) core.Spans {
+	spans := make([]core.Span, len(ids))
+	for i, id := range ids {
+		docID := base.MakeDataStoreKeyWithCollectionAndDocID(desc, id)
+		spans[i] = core.NewSpan(docID, docID.PrefixEnd())
 	}
-	n.doc = doc
-	return nil
+	return core.NewSpans(spans...)
 }
 
-// Next only returns once.
-func (n *createNode) Next() (bool, error) {
-	n.execInfo.iterations++
-
-	if n.err != nil {
-		return false, n.err
-	}
-
-	if n.returned {
-		return false, nil
-	}
-
-	if err := n.collection.Create(
-		n.p.ctx,
-		n.doc,
-	); err != nil {
-		return false, err
+func documentsToDocIDs(docs []*client.Document) []string {
+	docIDs := make([]string, len(docs))
+	for i, doc := range docs {
+		docIDs[i] = doc.ID().String()
 	}
+	return docIDs
+}
 
-	currentValue := n.documentMapping.NewDoc()
+func (n *createNode) Start() (err error) {
+	n.docs = make([]*client.Document, len(n.input))
+	defer func() { n.err = err }()
 
-	currentValue.SetID(n.doc.ID().String())
-	for i, value := range n.doc.Values() {
-		if len(n.documentMapping.IndexesByName[i.Name()]) > 0 {
-			n.documentMapping.SetFirstOfName(&currentValue, i.Name(), value.Value())
-		} else if aliasName := i.Name() + request.RelatedObjectID; len(n.documentMapping.IndexesByName[aliasName]) > 0 {
-			n.documentMapping.SetFirstOfName(&currentValue, aliasName, value.Value())
-		} else {
-			return false, client.NewErrFieldNotExist(i.Name())
+	for i, input := range n.input {
+		doc, err := client.NewDocFromMap(input, n.collection.Definition())
+		if err != nil {
+			return err
 		}
+		n.docs[i] = doc
 	}
 
-	n.returned = true
-	n.currentValue = currentValue
-
-	desc := n.collection.Description()
-	docID := base.MakeDataStoreKeyWithCollectionAndDocID(desc, currentValue.GetID())
-	n.results.Spans(core.NewSpans(core.NewSpan(docID, docID.PrefixEnd())))
-
-	err := n.results.Init()
-	if err != nil {
-		return false, err
+	if len(n.docs) == 1 {
+		err = n.collection.Create(n.p.ctx, n.docs[0])
+	} else {
+		err = n.collection.CreateMany(n.p.ctx, n.docs)
```

Comment:
todo: I do disagree with creating docs in `Start` - it feels like it is very much in the wrong place and may trip us up in the future (if not by being technically annoying for some future change, it seems likely to cause 'surprise').

Please either move this call into `Next`, or discuss in in discord/standup (for greater visibility).

Reply:
I understand your concern. I had similar feelings but thought it was highly unlikely and having a nicer code was worth it. If you think otherwise, I will move it to `Next`. No problem.

Reply:
Thanks Islam, sorry about the bother, and I understand it is a bit 'abstract' in this case, but I do think it is in the wrong place atm.  Ideally we would remove the Start/Init funcs anyway, and conceptually I very much think the 'creating' part of a 'create' iterator should be done in the iteration funcs (next/value).